app/models/library_entry.rb

```diff
@@ -164,13 +167,20 @@ def kind
   end
 
   before_save do
-    if status_changed? && completed? && media&.progress_limit
-      # When marked completed, we try to update progress to the cap
-      self.progress = media.progress_limit
-    elsif !status_changed? && progress == media&.progress_limit
-      # When in current and progress equals total episodes
-      self.status = :completed
+    if status_changed? && completed?
+      # update progress to the cap
+      self.progress = media.progress_limit if media&.progress_limit
+      # set finished_at and started_at for the first consume
+      self.finished_at ||= Time.now
+      self.started_at ||= Time.now
     end
+
+    # When progress equals total episodes
+    self.status = :completed if progress == media&.progress_limit
```

Comment:
This caused a bug previously where you couldn't rewatch, which is why we have `!status_changed?` to prevent it from triggering when the status changes iirc